src/sinks/redis.rs

```diff
@@ -0,0 +1,587 @@
+#[derivative(Default)]
+#[serde(rename_all = "lowercase")]
+pub enum DataType {
+    #[derivative(Default)]
+    List,
+    Channel,
+}
+
+#[derive(Copy, Clone, Debug, Derivative, Deserialize, Serialize, Eq, PartialEq)]
+#[derivative(Default)]
+#[serde(rename_all = "lowercase")]
+pub enum Method {
+    #[derivative(Default)]
+    LPush,
+    RPush,
+}
+
+#[derive(Copy, Clone, Debug, Derivative, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub struct ListOption {
+    method: Method,
+}
+
+#[derive(Clone, Copy, Debug, Derivative, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum Encoding {
+    Text,
+    Json,
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig<Option<usize>> = TowerRequestConfig {
+        rate_limit_num: Option::from(u64::MAX),
+        ..Default::default()
+    };
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct RedisSinkConfig {
+    encoding: EncodingConfig<Encoding>,
+    #[serde(default)]
+    data_type: DataType,
+    #[serde(alias = "list")]
+    list_option: Option<ListOption>,
+    url: String,
+    key: String,
+    #[serde(default)]
+    batch: BatchConfig,
+    #[serde(default)]
+    request: TowerRequestConfig<Option<usize>>,
+}
+
+impl GenerateConfig for RedisSinkConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"
+            url = "redis://127.0.0.1:6379/0"
+            key = "vector"
+            data_type = "list"
+            list.method = "lpush"
+            encoding.codec = "json"
+            batch.max_events = 1
+            "#,
+        )
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "redis")]
+impl SinkConfig for RedisSinkConfig {
+    async fn build(
+        &self,
+        cx: SinkContext,
+    ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
+        if self.key.is_empty() {
+            return Err("`key` cannot be empty.".into());
+        }
+        let conn = self.build_client().await.context(RedisCreateFailed)?;
+        let healthcheck = RedisSinkConfig::healthcheck(conn.clone()).boxed();
+        let sink = self.new(conn, cx)?;
+        Ok((sink, healthcheck))
+    }
+
+    fn input_type(&self) -> config::DataType {
+        config::DataType::Log
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "redis"
+    }
+}
+
+impl RedisSinkConfig {
+    pub fn new(
+        &self,
+        conn: ConnectionManager,
+        cx: SinkContext,
+    ) -> crate::Result<super::VectorSink> {
+        let request = self.request.unwrap_with(&REQUEST_DEFAULTS);
+
+        let key = Template::try_from(self.key.clone()).context(KeyTemplate)?;
+        let encoding = self.encoding.clone();
+
+        let mut method: Option<Method> = None;
+        if self.list_option.is_some() {
+            method = Option::from(self.list_option.unwrap().method)
+        }
```

Comment:
```suggestion
        let method = self.list_option.map(|option| option.method);
```

This will avoid the `unwrap`.